mozilla/treeherder
# Bug 1566190 - Implement Taskcluster login change

This pr decouples taskcluster credentials from Auth0, which is kept strictly for user login and session management. _For reasons mentioned below, please only review the new login and taskcluster credential flows._
How to test these changes:
1) Login in to Treeherder - you'll need to grant permission to treeherder.mozilla.org and login continues as usual.

2) Perform a taskcluster action that requires credentials, like triggering a job. A new window will open redirecting the user to the taskcluster staging instance, and requires a login with auth0 or github. This won't show every time and I think @helfi92 had mentioned that someone is working on streamlining this for firefox-ci. After credentials are saved the window will close. 

3) You'll also notice the TC action probably won't work because the staging instance doesn't have the same tasks as taskcluster.net so performing an action with the new credentials won't work. (Note: Hassan did set up an example task that I successfully queried with the new credentials)

4) If you try to perform an action again, it shouldn't open the window to retrieve credentials unless they've expired.

I'm not planning on merging this until the taskcluster.net switch off on Nov 9th. I'll coordinate with the Taskcluster team so I can update this pr with the new rootUrls and have sheriffs test all Taskcluster actions before doing a quick merge/prod push. We'll probably also want to give Sheriff's a heads up about the likely down time between when taskcluster.net is shut off and the new instances are active (and due to pulse ingestion with the new rootUrls).

I've modeled the `TaskClusterCallback` component after the `LoginCallback` component and updated the messaging to keep it consistent for both. I've launched this to prototype for testing. 

@helfi92 or @djmitche if one of you would like to give this a look that'd be appreciated.

FYI @karlht 

Edit: If people are happy with this flow, I'll see about adding a few tests.
